### PR TITLE
Show function aliases as function declarations in completions

### LINKED ISSUE #25: Functions defined directly as a field access of an import currently complete as variables
Quick example:  
```zig
const std = @import("std");

pub fn main() void {
    std.ArrayLi // <- ArrayList completion is a variable, not a function
}

```

### PR DESCRIPTION
**Pull request by Sergeeeek**
Resolves #25

### CODE REVIEW

File: src/document_store.zig [hunk 1]
```diff
@@ -248,6 +252,8 @@ fn uriFromImportStr(store: *DocumentStore, allocator: *std.mem.Allocator, handle
 }
 
 pub const AnalysisContext = struct {
+    const Self = @This();
```
Comment (alexnask) [inline src/document_store.zig:255]: Minor style point, I don't think this is needed, you can use `AnalysisContext` in `clone`

File: src/main.zig [hunk 2]
```diff
@@ -340,11 +355,14 @@ fn completeGlobal(id: i64, pos_index: usize, handle: DocumentStore.Handle, confi
     // Deallocate all temporary data.
     defer arena.deinit();
 
+    var analysis_ctx = try document_store.analysisContext(handle, &arena, types.Position{.line = 0, .character = 0,});
```
Comment (alexnask) [inline src/main.zig:358]: Can you remove the trailing comma in the position here?  
This will cause zig fmt to make it multiline

**Comment by alexnask** [comment]
This looks very nice, only a minor style points to fix in my opinion.

File: src/document_store.zig (hunk 1 above)
Comment (Sergeeeek) [reply src/document_store.zig]: Fixed

File: src/main.zig (hunk 2 above)
Comment (Sergeeeek) [reply src/main.zig]: zig fmt failed for some reason and I didn't notice, fixed now :)

**Comment by alexnask** [comment]
Cool, thanks for the great PR :) 👍

**Comment by Sergeeeek** [comment]
Thank you! Took me long enough, haha

**Comment by alexnask** [comment]
This will cause us to import a lot of stuff when first using std, since it includes functions from these files.  
Not a bad thing, just an observation :-) 

```
Standard library base uri: file:///c%3A/dev/zig/build-release/lib/zig/std/std.zig
------------------------------------------ Allocation info ------------------------------------------
14 total allocations (total: 0.13 MB, mean: 0.01 MB, std. dev: 0.01 MB), 11 deallocations
3 current allocations (0.00 MB)
6 reallocations (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
5 shrinks (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
-----------------------------------------------------------------------------------------------------
------------------------------------------ Allocation info ------------------------------------------
19 total allocations (total: 0.13 MB, mean: 0.01 MB, std. dev: 0.01 MB), 16 deallocations
3 current allocations (0.00 MB)
7 reallocations (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
7 shrinks (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
-----------------------------------------------------------------------------------------------------
Opened document: file:///c%3A/dev/test2.zig
------------------------------------------ Allocation info ------------------------------------------
32 total allocations (total: 0.16 MB, mean: 0.01 MB, std. dev: 0.01 MB), 25 deallocations
7 current allocations (0.00 MB)
8 reallocations (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
8 shrinks (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
-----------------------------------------------------------------------------------------------------
New text for document file:///c%3A/dev/test2.zig
------------------------------------------ Allocation info ------------------------------------------
40 total allocations (total: 0.19 MB, mean: 0.00 MB, std. dev: 0.01 MB), 33 deallocations
7 current allocations (0.00 MB)
9 reallocations (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
9 shrinks (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
-----------------------------------------------------------------------------------------------------
New text for document file:///c%3A/dev/test2.zig
------------------------------------------ Allocation info ------------------------------------------
49 total allocations (total: 0.21 MB, mean: 0.00 MB, std. dev: 0.01 MB), 42 deallocations
7 current allocations (0.00 MB)
10 reallocations (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
10 shrinks (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
-----------------------------------------------------------------------------------------------------
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/std.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/std.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/std.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/json.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/json.zig
INVALID: Id.InfixOp
INVALID: Id.SuffixOp
------------------------------------------ Allocation info ------------------------------------------
106 total allocations (total: 2.08 MB, mean: 0.02 MB, std. dev: 0.07 MB), 89 deallocations
17 current allocations (0.10 MB)
14 reallocations (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
16 shrinks (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
-----------------------------------------------------------------------------------------------------
New text for document file:///c%3A/dev/test2.zig
------------------------------------------ Allocation info ------------------------------------------
118 total allocations (total: 2.13 MB, mean: 0.02 MB, std. dev: 0.07 MB), 101 deallocations
17 current allocations (0.10 MB)
15 reallocations (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
17 shrinks (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
-----------------------------------------------------------------------------------------------------
New text for document file:///c%3A/dev/test2.zig
------------------------------------------ Allocation info ------------------------------------------
130 total allocations (total: 2.17 MB, mean: 0.02 MB, std. dev: 0.07 MB), 113 deallocations
17 current allocations (0.10 MB)
16 reallocations (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
18 shrinks (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
-----------------------------------------------------------------------------------------------------
New text for document file:///c%3A/dev/test2.zig
------------------------------------------ Allocation info ------------------------------------------
142 total allocations (total: 2.22 MB, mean: 0.02 MB, std. dev: 0.06 MB), 125 deallocations
17 current allocations (0.10 MB)
17 reallocations (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
19 shrinks (total: 0.00 MB, mean: 0.00 MB, std. dev: 0.00 MB)
-----------------------------------------------------------------------------------------------------
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/std.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/array_list.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/array_list.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/array_list.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/array_list.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/array_list_sentineled.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/array_list_sentineled.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/array_list.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/hash_map.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/hash_map.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/bloom_filter.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/bloom_filter.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/buf_map.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/buf_map.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/buf_set.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/buf_set.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/child_process.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/child_process.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/dynamic_library.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/dynamic_library.zig
Type resolution case not implemented; Id.Switch
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/hash_map.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/mutex.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/mutex.zig
Type resolution case not implemented; Id.If
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/packed_int_array.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/packed_int_array.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/packed_int_array.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/packed_int_array.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/packed_int_array.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/priority_queue.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/priority_queue.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/progress.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/progress.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/reset_event.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/reset_event.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/segmented_list.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/segmented_list.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/linked_list.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/linked_list.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/spinlock.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/spinlock.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/hash_map.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/linked_list.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/target.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/target.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/thread.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/thread.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/atomic.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/atomic.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/base64.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/base64.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/build.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/build.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/builtin.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/builtin.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/c.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/c.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/coff.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/coff.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/crypto.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/crypto.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/cstr.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/cstr.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/debug.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/debug.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/dwarf.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/dwarf.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/elf.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/elf.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/event.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/event.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/fifo.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/fifo.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/fmt.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/fmt.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/fs.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/fs.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/hash.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/hash.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/hash_map.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/heap.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/heap.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/http.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/http.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/io.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/io.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/json.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/macho.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/macho.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/math.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/math.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/mem.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/mem.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/meta.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/meta.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/net.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/net.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/os.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/os.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/once.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/once.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/packed_int_array.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/pdb.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/pdb.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/process.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/process.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/rand.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/rand.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/rb.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/rb.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/sort.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/sort.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/ascii.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/ascii.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/testing.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/testing.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/time.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/time.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/unicode.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/unicode.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/valgrind.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/valgrind.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/zig.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/zig.zig
Import final URI: file:///c%3A/dev/zig/build-release/lib/zig/std/start.zig
Opened document: file:///c%3A/dev/zig/build-release/lib/zig/std/start.zig
INVALID: Id.Comptime
INVALID: Id.TestDecl
------------------------------------------ Allocation info ------------------------------------------
1408 total allocations (total: 35.77 MB, mean: 0.03 MB, std. dev: 0.06 MB), 1167 deallocations
241 current allocations (1.46 MB)
232 reallocations (total: 0.01 MB, mean: 0.00 MB, std. dev: 0.00 MB)
278 shrinks (total: 0.01 MB, mean: 0.00 MB, std. dev: 0.00 MB)
-----------------------------------------------------------------------------------------------------
```

Memory usage is very low even importing all this stuff.

**Comment by SuperAuguste** [comment]
Looks good!

**Pull request merged**